core/opendaq/component/include/opendaq/folder_impl.h

```diff
@@ -163,54 +167,103 @@ ErrCode FolderImpl<Intf, Intfs...>::addItem(IComponent* item)
 {
     OPENDAQ_PARAM_NOT_NULL(item);
 
-    std::scoped_lock lock(this->sync);
+    {
+        std::scoped_lock lock(this->sync);
 
-    return daqTry(
-        [this, &item]
-        {
-            if (!addItemInternal(item))
-                return OPENDAQ_ERR_DUPLICATEITEM;
+        const ErrCode err = daqTry(
+            [this, &item]
+            {
+                if (!addItemInternal(item))
+                    return OPENDAQ_ERR_DUPLICATEITEM;
+
+                return OPENDAQ_SUCCESS;
+            });
+
+        if (OPENDAQ_FAILED(err))
+            return err;
+    }
 
-            return OPENDAQ_SUCCESS;
-        });
+    if (!this->coreEventMuted && this->coreEvent.assigned())
+    {
+        const auto component = ComponentPtr::Borrow(item);
+        const auto args = createWithImplementation<ICoreEventArgs, CoreEventArgsImpl>(
```

Comment:
What is the reason for not using factories? Doesn't that force that CoreEventArgsImpl has to be header-only implementation? If true this increases compilation time of modules.

Reply:
Issues with dependencies. Events are called from within the Component implementation, but the factories require a dependency to signals. So we'd need to have several headers for factory definitions in the opendaq project. 

So I suppose it's a issue of compile time vs. organization clarity.